src/components/Jam.jsx

```diff
@@ -40,14 +45,65 @@ export default function Jam() {
   const [collaborativeOperationRunning, setCollaborativeOperationRunning] = useState(false)
   const [schedule, setSchedule] = useState(null)
 
+  const getNewAddresses = useCallback(
+    (count, mixdepth) => {
+      if (!walletInfo) {
+        throw new Error('Wallet info is not available.')
+      }
+      const externalBranch = walletInfo.data.display.walletinfo.accounts[mixdepth].branches.find((branch) => {
+        return branch.branch.split('\t')[0] === 'external addresses'
+      })
+
+      const newAddresses = []
+
+      externalBranch.entries.every((entry) => {
+        if (entry.status === 'new') {
+          newAddresses.push(entry.address)
+        }
+
+        if (newAddresses.length >= count) {
+          return false
+        }
+
+        return true
+      })
+
+      if (newAddresses.length !== count) {
+        throw new Error(`Cannot find requested amount of addresses: found ${newAddresses.length} of ${count}`)
+      }
+
+      return newAddresses
+    },
+    [walletInfo]
+  )
+
   // Todo: Testing toggle is deactivated until https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1260 is merged.
   const deactivateTestingToggle = true
   const [useInsecureTestingSettings, setUseInsecureTestingSettings] = useState(false)
 
-  // Todo: Discuss if we should hardcode this or let the user pick an account.
-  const INTERNAL_DEST_ACCOUNT = 0
-  // Interval in milliseconds between requests to reload the schedule.
-  const SCHEDULE_REQUEST_INTERVAL = process.env.NODE_ENV === 'development' ? 10_000 : 60_000
+  const initialFormValues = useMemo(() => {
+    const addressCount = 3
+
+    let destinationAddresses = []
+    if (destinationIsExternal) {
+      // prefill with empty addresses
+      destinationAddresses = Array(addressCount).fill('')
+    } else {
+      try {
+        // prefill with addresses marked as "new"
+        destinationAddresses = getNewAddresses(addressCount, INTERNAL_DEST_ACCOUNT)
+      } catch (e) {
+        // on error initialize with empty addresses - form validation will do the rest
+        destinationAddresses = Array(addressCount).fill('')
+      }
+    }
+
+    const destinationAddressFormMap = destinationAddresses.reduce(
+      (obj, addr, index) => ({ ...obj, [`dest${index + 1}`]: addr }),
+      {}
+    )
+    return { ...destinationAddressFormMap }
```

Comment:
I'd just return the result of `reduce` directly. Then we don't need to come up with a variable name and spread it out again the next line. :)